### ISSUE TITLE
Timestamp improvement

### ISSUE THREAD

**Issue by usernotnull** [platers/obsidian-linter#221]
I have linting set to auto-lint on save.

As a habit, I always lint files, even if I did not modify. The issue is if there was no linting issue (the file was not changed), the modified date is updated regardless, even if nothing was actually modified.

It gets worse when I lint all files, because all files will have their modified timestamps changed. This is an issue esp when using version control, because this will modify all files all the time.

NB: I don't think anyone would want to change the timestamps of an unchanged file, so maybe it would be better to just add this improvement without a settings checkbox. Should be a minor version change.

**Comment by pjkaufman** [comment]
I am curious as to whether it actually updates the date modified on the file. If that is the case, the update of the date modified is accurate. However if it does not match the date modified on the file, I can see what your saying being the case.

Are you able to verify whether the date modified matches the file metadata, @usernotnull ?

**Comment by pjkaufman** [comment]
From what I can tell, this is a bug assuming it is happening. I just need to reproduce the scenario and then I can see about fixing this.

**Comment by usernotnull** [comment]
Ok, so I usually lint by pressing CTRL + S to save & lint. I assumed the issue may be from obsidian since it might be rewriting the file even if there are no changes.

But I also tried with the command Lint current file, and the same thing is happening, the YAML modified date always changes to the current date too even if there are no changes.

This also applies to lint all file in vault command.

The modified date YAML on lint is the current date of command execution. It will coincide with the file modified date since the file is actually modified.

I hope I understood and answered your question.

**Comment by pjkaufman** [comment]
This does seem like a bug to me, but it will take some time to determine how to properly do this given the order rules run in is not defined.

**Comment by usernotnull** [comment]
Were you able to reproduce the issue?

**Comment by pjkaufman** [comment]
I was not able to yet, but I did confirm that just running lint file/ctrl+s does not update the date updated unless a change is present in the file.

**Comment by usernotnull** [comment]
I'm running the latest obsidian on Windows 11.

I created a new vault, added only the linter plugin.

In the settings of linter, I have enabled to `keep track of the date the file was last edited in the YAML front matter`.

Now, every time I run `Lint the current file`, Linter updates the `date modified` to the current date. 

Now, this could be your intended use of that rule, to actually always rewrite the modified date to the time when Linter runs. 
But I'm suggesting to see if the modified date needs to be updated if nothing was actually linted in the file (as in, if no other rule was actually applied and the original file not changed by linter, the modified rule date should not be applied).
